api/provider/aws/service_update.go

```diff
@@ -0,0 +1,70 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (s *ServiceProvider) Update(req models.UpdateServiceRequest) error {
+	serviceID := req.ServiceID
+	deployID := req.DeployID
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	serviceTags, err := s.TagStore.SelectByTypeAndID("service", serviceID)
+	if err != nil {
+		return err
+	}
+
+	tag, ok := serviceTags.WithKey("environment_id").First()
+	if !ok {
+		// is this canonical error handling?
+		return fmt.Errorf("Cannot resolve environment_id for service %s", serviceID)
+	}
+
+	environmentID := tag.Value
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+
+	service, err := s.readService(clusterName, serviceID)
+	if err != nil {
+		return err
+	}
+
+	var serviceScaleCount *int64
```

Comment:
I think we should follow a more general pattern for all updates:
```
if req.OptionA != nil {
    updateOptionA(...)
}

if req.OptionB != nil {
    updateOptionB(...)
}

...

return nil
```

In that regard, I think we have a cleaner, more readable solution (even though it's a bit longer) by splitting out `updateServiceScale` and `updateServiceTaskDefinition` into different helper functions. 

That also lets us avoid calling `Read()` inside this function.

Reply:
We're okay with unnecessarily making two `s.AWS.ECS.UpdateService(input)` calls when a user wants to update both scale and deploy at once? It doesn't seem like that would cause any issues, but I've tried to keep an eye on how many API calls we make since Jonathan et alia have been having rate limit problems.

Then again, I'm already making two AWS calls; one for reading, and one for updating. Stripping out the reading call in favor adding a _potential_ second update call is an improvement. 

Cool, I'm onboard. :+1:

Reply:
The rate of aws calls is a good thing to keep in mind, but the create/update/delete calls pale in comparison to read and list calls.

Reply:
✅